src/Models/Category.php

```diff
@@ -15,6 +15,6 @@ class Category extends Model
 
     public function getTable()
     {
-        return 'catalog_category_flat_store_'.(Statamic::isCpRoute() ? (Site::selected()->attributes['magento_store_id'] ?? '1') : (Site::current()->attributes['magento_store_id'] ?? '1'));
+        return 'catalog_category_flat_store_'.once(fn() => (Statamic::isCpRoute() ? (Site::selected()->attributes['magento_store_id'] ?? '1') : (Site::current()->attributes['magento_store_id'] ?? '1')));
```

Comment:
Is this going as expected when switching between de frontend and the backend? And switching sites from the backend?

Reply:
It should, since once only memoizes the data per response. Switching store would use another request, thus no cache the first time.
The only case where this could be an issue is while switching store within the same request which doesn't happen.
(We do switch the store within the same request in rapidez:index but that does not impact this model)